### PR TITLE
use css pixels for playwright screenshot, add new focusAndInputLocation command, bump playwright min version to 1.21

### PR DESCRIPTION
**Pull request by khsheehan**
use css pixels for playwright screenshot, add new focusAndInputLocation command, bump playwright min version to 1.21

### CODE REVIEW

File: packages/playwright/src/cdp.ts [hunk 1]
```diff
@@ -62,12 +62,6 @@ export const get = async (page: Page, args: { url: string }) => {
   })
 }
```
Comment (khsheehan) [inline packages/playwright/src/cdp.ts:64]: Unused, won't be needed

File: packages/playwright/src/cdp.ts [hunk 2]
```diff
@@ -163,7 +157,7 @@ export const clickElement = async (page: Page, args: { id: string }) => {
   return true
 }
```
Comment (khsheehan) [inline packages/playwright/src/cdp.ts:159]: Playwright no longer makes it easy to pluck out these Protocol types. This is still correctly typed

File: packages/playwright/src/index.ts [hunk 3]
```diff
@@ -148,6 +148,7 @@ const runCommandsToCompletion = async (page: Page, test: TestType<any, any>, tas
  */
 const executeCommand = async (page: Page, command: CommandRequestZeroStepMessage): Promise<any> => {
   switch (command.name) {
```
Comment (khsheehan) [inline packages/playwright/src/index.ts:150]: clickAndInputLocation is new, everything else just reorganized for future

File: packages/playwright/src/playwright.ts [hunk 4]
```diff
@@ -17,17 +17,38 @@ export const hoverElement = async (page: Page, args: { id: string }) => {
...
 }
 
-export const clickLocation = async (page: Page, args: { x: number, y: number }) => {
-  await hoverLocation(page, args)
-  await page.mouse.click(args.x, args.y)
-}
-
 export const sendKeys = async (page: Page, args: { value: string }) => {
   await page.keyboard.type(args.value)
 }
```
Comment (khsheehan) [inline packages/playwright/src/playwright.ts:23]: These location-based commands now use locators

File: packages/playwright/src/playwright.ts [hunk 5]
```diff
@@ -43,7 +64,7 @@ export const getViewportMetadata = async (page: Page) => {
...
 
 export const getScreenshot = async (page: Page) => {
-  const buffer = await page.screenshot()
+  const buffer = await page.screenshot({ scale: 'css' })
```
Comment (khsheehan) [inline packages/playwright/src/playwright.ts:67]: Single pixel for each css pixel

**Review by MitchHub — APPROVED** [review]

**Pull request merged**